File: reverse-prefix.js
/**
 * Given a string word and a character ch, reverses the segment of word
 * that starts at index 0 and ends at the index of the first occurrence
 * of ch (inclusive), and returns a new string containing the reversed
 * segment along with the rest of the word.
 * 
 * If ch does not exist in the word, returns the original string.
 * 
 * @param {string} word - A string parameter
 * @param {string} ch - A single string character
 * @returns {string} - Returns a new string
 */
function reversePrefix (word, ch) {
  let res = '';
  if (!word.includes(ch)) {
    res = word;
  } else {
    let index = word.indexOf(ch);
    let end = word.substring(index + 1);
    while (index >= 0) {
      res += word[index];
      index--;
    }
    res += end;
  }
  return res;
}

File: src/algorithms/reverse-prefix.py
def reverse_prefix(word, ch):
    """Given a 0-indexed string word and a character ch, reverse the segment of 
    word that starts at index 0 and ends at the index of the first occurrence 
    of ch (inclusive). If the character ch does not exist in word, do nothing.
    For example, if word = "abcdefd" and ch = "d", then you should reverse the 
    segment that starts at 0 and ends at 3 (inclusive). The resulting string 
    will be "dcbaefd". Return the resulting string.

    Args:
        word (str): A string of lowercase English letters.
        ch (str): A lowercase English letter.

    Returns:
        str: Returns a modified string.
    """
    index = -1
    for i in range(len(word)):
        if word[i] == ch:
            index = i
            break
    if index == -1:
        return word
    return word[index:: -1] + word[index + 1::]
